src/Parser/Mage/Fire/Modules/Features/CombustionMarqueeBindings.js

```diff
@@ -0,0 +1,159 @@
+  //Accounts for the extra 2 seconds from Tier 21 2 Set, and 1 Second per point of Pre Ignited
+  on_initialized() {
+    this.active = this.combatants.selected.hasWrists(ITEMS.MARQUEE_BINDINGS_OF_THE_SUN_KING.id);
+    const hasTierBonus = this.combatants.selected.hasBuff(SPELLS.FIRE_MAGE_T21_2SET_BONUS_BUFF.id);
+    const preIgnitedCount = this.combatants.selected.traitsBySpellId[SPELLS.PRE_IGNITED_TRAIT.id];
+    if (hasTierBonus) {
+      this.combustionDuration = (10 + 2 + preIgnitedCount) * 1000;
+    } else {
+      this.combustionDuration = (10 + preIgnitedCount) * 1000;
+    }
+  }
+
+  //Get the time stamp for Pyroblast to be used elsewhere
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id) {
+      return;
+    }
+    this.pyroblastCastTimestamp = event.timestamp;
+  }
+
+  //Check to see if the Bracer Proc used during Combustion landed while Combustion was still active.
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PYROBLAST.id || !this.buffUsedDuringCombustion) {
+      return;
+    }
+    this.buffUsedDuringCombustion = false;
+    if (!this.combatants.selected.hasBuff(SPELLS.COMBUSTION.id)) {
+      this.didntLandDuringCombustion += 1;
+      debug && console.log("Pyroblast Did Not Land During Combustion @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    }
+  }
+
+  on_toPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id && spellId !== SPELLS.COMBUSTION.id) {
+      return;
+    }
+    if (spellId === SPELLS.COMBUSTION.id) {
+      this.combustionCastTimestamp = event.timestamp;
+      //If the player had a Bracer Proc when Combustion was cast, then its expected for them to cast it during Combustion.
+      if (this.combatants.selected.hasBuff(SPELLS.KAELTHAS_ULTIMATE_ABILITY.id)) {
+        this.expectedPyroblastCasts += 1;
+        debug && console.log("Pyroblast Expected During Combustion @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      }
+    } else {
+      //If the player gets a Bracer Proc, and there is more than 5 seconds left on the duration of Combustion, then its expected for them to cast it during Combustion.
+      if (this.combustionEndTime - event.timestamp > 5000) {
+        this.expectedPyroblastCasts += 1;
+        debug && console.log("Pyroblast Expected During Combustion @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+      }
+    }
+  }
+
+  //Check to see if the player used their Bracer Proc during Combustion
+  on_toPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+      return;
+    }
+    if (event.timestamp - 50 < this.pyroblastCastTimestamp && this.combatants.selected.hasBuff(SPELLS.COMBUSTION.id)) {
```

Comment:
this isnt checking to see if they had combustion within the last 50 ms, its checking to see if pyroblast was cast within the last 50ms ... to confirm that they hard casted pyroblast and thats why the buff was removed instead of the buff expiring.

Reply:
ah, my bad.